### PR TITLE
Fix/cm policy

### LINKED ISSUE #533: `roleName` ODAHU deployment changes doesn't take effect.
**Description**
Deployment `roleName` field update doesn't actually trigger changes.

**Steps to Reproduce**
1. Create ODAHU deployment
2. Go to deployment page
3. Click `YAML` tab
4. Change `roleName` field value
5. Click `Submit` 

**Actual result**
Knative configuration doesn't change, revision is not created, pod doesn't restart, Authentication works via role that was previously in configuration.

**Expected result**
Knative configuration entity should change, then new model deployment revision should be created and new pod should be deployed. The new pod should authenticate via provided role. UI shows that deployment is ready and the new role name is displayed in the deployment configuration window and YAML.

**Environment:**
 - k8s version: 1.17
 - ODAHU version: 1.5.0-rc2

### PR DESCRIPTION
**Pull request by vlad-tokarev**
fix #533

### CODE REVIEW

File: packages/operator/controllers/modeldeployment_controller.go [hunk 1]
```diff
@@ -240,26 +244,41 @@ func (r *ModelDeploymentReconciler) ReconcileKnativeService(
...
-	err = r.Update(context.TODO(), found)
-	if err != nil {
-		return err
-	}
 
+	if depSpecChanged || policyIsChanged {
+		if policyIsChanged {
+			log.Info("Policy hash was changed",
+				"old", oldPolicyHash,
+				"new", newPolicyHash,
```
Comment (BPylypenko) [inline packages/operator/controllers/modeldeployment_controller.go:260]:
```suggestion
			log.Info("Policy hash was changed,",
				"old:", oldPolicyHash,
				"new:", newPolicyHash,
```

File: packages/operator/controllers/modeldeployment_controller.go [hunk 2]
```diff
@@ -240,26 +244,41 @@ func (r *ModelDeploymentReconciler) ReconcileKnativeService(
...
+			)
+		}
+		if depSpecChanged {
+			log.Info("ModelDeployment spec was changed","old", lastAppliedMDSpec,
+				"new", modelDeploymentCR.Spec)
```
Comment (BPylypenko) [inline packages/operator/controllers/modeldeployment_controller.go:265]:
```suggestion
		if depSpecChanged {
			log.Info("ModelDeployment spec was changed,","old:", lastAppliedMDSpec,
				"new:", modelDeploymentCR.Spec)
```

File: packages/operator/controllers/modeldeployment_controller.go [hunk 3]
```diff
@@ -240,26 +244,41 @@ func (r *ModelDeploymentReconciler) ReconcileKnativeService(
...
+		}
+
+		err = fulfilKnativeService(found)
+		if err != nil {
+			return err
+		}
+		log.Info(fmt.Sprintf("Updating %s Knative Service, new MD generation: %d",
```
Comment (BPylypenko) [inline packages/operator/controllers/modeldeployment_controller.go]:
```suggestion
		log.Info(fmt.Sprintf("Updating '%s' Knative Service, new MD generation: %d",
```

**Review by BPylypenko — APPROVED** [review]

File: packages/operator/controllers/modeldeployment_controller.go (hunk 2 above)
Comment (vlad-tokarev) [reply packages/operator/controllers/modeldeployment_controller.go:265]: these key values are set automatically with : as a separator

**Review by keshamin — APPROVED** [review]
As discussed in chat, please, complete the logic for case when no role is set. The label with policy config map is to be removed.

**Review by easokol — APPROVED** [review]

**Pull request merged**